test/filters/test_crop_hull.cpp

```diff
@@ -0,0 +1,586 @@
+            return p;
+          };
+          auto outsidePointGenerator = [this, &baseOffset] () {
+            pcl::PointXYZ p(rd(gen) + 2., rd(gen) + 2., rd(gen) + 2.);
+            p.getVector3fMap() += baseOffset.getVector3fMap();
+            return p;
+          };
+          pcl::CropHull<pcl::PointXYZ> cropHullFilter = createDefaultCropHull(inputCloud);
+          cropHullFilter.setDim(3);
+          std::vector<TestData> testDataSuite = createTestDataSuite(insidePointGenerator, outsidePointGenerator);
+          for (TestData const & testData : testDataSuite)
+          {
+            checker(cropHullFilter, testData);
+          }
+        }
+      }
+    private:
+      std::mt19937 gen;
+      std::uniform_real_distribution<float> rd;
+      pcl::PointCloud<pcl::PointXYZ>::Ptr baseInputCloud;
+  };
+}
+
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+TEST_F (PCLCropHullTestFixture2d, 2dsquare_simple_test)
+{
+  auto checker = [](pcl::CropHull<pcl::PointXYZ> cropHullFilter, TestData const & testData) {
+    cropHullFilter.setInputCloud(testData.inputCloud);
+    pcl::Indices filteredIndices;
+    cropHullFilter.filter(filteredIndices);
+    pcl::test::EXPECT_EQ_VECTORS(testData.insideIndices, filteredIndices);
+  };
+  test(checker);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+TEST_F (PCLCropHullTestFixture2d, 2dsquare_test_reentrancy)
+{
+  auto checker = [](pcl::CropHull<pcl::PointXYZ> cropHullFilter, TestData const & testData) {
+    cropHullFilter.setInputCloud(testData.inputCloud);
+    pcl::Indices filteredIndices;
+    cropHullFilter.filter(filteredIndices);
+    pcl::test::EXPECT_EQ_VECTORS(testData.insideIndices, filteredIndices);
+    cropHullFilter.filter(filteredIndices);
+    pcl::test::EXPECT_EQ_VECTORS(testData.insideIndices, filteredIndices);
+  };
+  test(checker);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+TEST_F (PCLCropHullTestFixture2d, 2dsquare_test_negative_filter_functionality)
+{
+  auto checker = [](pcl::CropHull<pcl::PointXYZ> cropHullFilter, TestData const & testData) {
+    cropHullFilter.setInputCloud(testData.inputCloud);
+    cropHullFilter.setNegative(true);
+    pcl::Indices filteredIndices;
+    cropHullFilter.filter(filteredIndices);
+    pcl::test::EXPECT_EQ_VECTORS(testData.outsideIndices, filteredIndices);
+    pcl::test::EXPECT_EQ_VECTORS(testData.insideIndices, *cropHullFilter.getRemovedIndices());
+  };
+  test(checker);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+TEST_F (PCLCropHullTestFixture2d, 2dsquare_test_crop_inside)
+{
+  auto checker = [](pcl::CropHull<pcl::PointXYZ> cropHullFilter, TestData const & testData) {
+    cropHullFilter.setInputCloud(testData.inputCloud);
```

Comment:
This line can be moved one step up in the call stack (`test` function). Common in every checker

Reply:
applied suggestion